### PR TITLE
[SRVOCF-437] Functions docs - update links, add substep

### PR DESCRIPTION
**Pull request by rh-max**
Version(s): 4.6 and 4.8+

Issue: https://issues.redhat.com/browse/SRVOCF-437

Link to docs preview: https://51030--docspreview.netlify.app/openshift-enterprise/latest/serverless/functions/serverless-functions-on-cluster-builds.html#serverless-functions-creating-on-cluster-builds_serverless-functions-on-cluster-builds

Dev & QE reviews: complete

### CODE REVIEW

**Comment by rh-max** [comment]
@lance @jrangelramos Hi, could you PTAL? Thank you.

**Review by lance — APPROVED** [review]
/lgtm

File: modules/serverless-functions-on-cluster-builds.adoc [hunk 1]
```diff
@@ -27,14 +27,14 @@ include::snippets/technology-preview.adoc[leveloffset=+1]
 +
 [source,terminal]
 ----
-$ oc apply -f https://raw.githubusercontent.com/openshift-knative/kn-plugin-func/serverless-1.22.0/pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml
+$ oc apply -f https://raw.githubusercontent.com/openshift-knative/kn-plugin-func/serverless-1.24.0/pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml
```
Comment (jrangelramos) [inline modules/serverless-functions-on-cluster-builds.adoc:30]: Here it needs to be 1.25.0

File: modules/serverless-functions-on-cluster-builds.adoc [hunk 2]
```diff
@@ -27,14 +27,14 @@ include::snippets/technology-preview.adoc[leveloffset=+1]
...
 ----
```
Comment (jrangelramos) [inline modules/serverless-functions-on-cluster-builds.adoc:31]: Here we need to include the s2i task as well

```
$ oc apply -f https://raw.githubusercontent.com/openshift-knative/kn-plugin-func/serverless-1.25.0/pipelines/resources/tekton/task/func-s2i/0.1/func-s2i.yaml
```

File: modules/serverless-functions-on-cluster-builds.adoc [hunk 3]
```diff
@@ -27,14 +27,14 @@ include::snippets/technology-preview.adoc[leveloffset=+1]
...
 
 .. Create the `kn func` deploy Tekton task to be able to deploy the function in the pipeline:
 +
 [source,terminal]
 ----
-$ oc apply -f https://raw.githubusercontent.com/openshift-knative/kn-plugin-func/serverless-1.22.0/pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml
+$ oc apply -f https://raw.githubusercontent.com/openshift-knative/kn-plugin-func/serverless-1.24.0/pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml
```
Comment (jrangelramos) [inline modules/serverless-functions-on-cluster-builds.adoc:37]: Same here.. it should be 1.25.0

```suggestion
$ oc apply -f https://raw.githubusercontent.com/openshift-knative/kn-plugin-func/serverless-1.25.0/pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml
```

**Review by jrangelramos — COMMENTED** [review]
Hello @rh-max. Here, the links should point to 1.25.0 and we need to include the s2i image task as well.

**Review by jrangelramos — COMMENTED** [review]
Hello @rh-max . just one additional comment is that the `build: git`
is not longer tracked on `func.yaml` (user now need to pass the `--remote`), so you can remove the lines 54 and 61 from serverless-functions-on-cluster-builds.adoc and include `--remote` on the line 72 (`kn func deploy --remote`)

File: modules/serverless-functions-on-cluster-builds.adoc (hunk 2 above)
Comment (rh-max) [reply modules/serverless-functions-on-cluster-builds.adoc:31]: Added, can you please check the wording? https://github.com/openshift/openshift-docs/pull/51030/commits/31b4677cc44b8d6e4bcfdf405b50397bfd8fcd04

**Comment by rh-max** [comment]
Thanks for the reviews @jrangelramos @lance !

> Hello @rh-max . just one additional comment is that the `build: git` is not longer tracked on `func.yaml` (user now need to pass the `--remote`), so you can remove the lines 54 and 61 from serverless-functions-on-cluster-builds.adoc and include `--remote` on the line 72 (`kn func deploy --remote`)

Done (I mis-committed, and the changes are split between here and https://github.com/openshift/openshift-docs/pull/51183 , but they are there).

Only awaiting one wording check from you. Thanks.

File: modules/serverless-functions-on-cluster-builds.adoc (hunk 2 above)
Comment (rh-max) [reply modules/serverless-functions-on-cluster-builds.adoc:31]: Actually, use this link instead: https://github.com/openshift/openshift-docs/pull/51030/commits/31b4677cc44b8d6e4bcfdf405b50397bfd8fcd04#diff-3f93bad3a6dccebfacff22389de39b99cf910accf4cadd053414fd97349accd0R40

**Comment by jrangelramos** [comment]
QE Approved

File: modules/serverless-functions-on-cluster-builds.adoc [hunk 4]
```diff
@@ -51,17 +58,15 @@ $ kn func create <function_name> -l <runtime>
...
+  revision: main <2>
+  contextDir: <directory_path> <3>
 ...
 ----
-<1> Required. Specify `git` build type.
-<2> Required. Specify the Git repository that contains your function's source code.
-<3> Optional. Specify the Git repository revision to be used. This can be a branch, tag or commit.
-<4> Optional. Specify the function's directory path if the function is not located in the Git repository root folder.
+<1> Required. Specify the Git repository that contains your function's source code.
+<2> Optional. Specify the Git repository revision to be used. This can be a branch, tag or commit.
```
Comment (Amrita42) [inline modules/serverless-functions-on-cluster-builds.adoc:68]: tag >> tag,

File: modules/serverless-functions-on-cluster-builds.adoc [hunk 5]
```diff
@@ -27,14 +27,21 @@ include::snippets/technology-preview.adoc[leveloffset=+1]
 +
 [source,terminal]
 ----
```
Comment (matejvasek) [inline modules/serverless-functions-on-cluster-builds.adoc:29]: wrt buildpack we should mention it is not supported -- pure upstream

**Comment by jrangelramos** [comment]
QE Approved

File: modules/serverless-functions-on-cluster-builds.adoc (hunk 5 above)
Comment (rh-max) [reply modules/serverless-functions-on-cluster-builds.adoc:29]: @matejvasek We removed the buildpacks step from here and put it inside a note that specifically states that they are not supported:

https://github.com/openshift/openshift-docs/pull/51030/commits/f4b292d06d1640408cebee3541be1cb8cf2fe4cf#diff-3f93bad3a6dccebfacff22389de39b99cf910accf4cadd053414fd97349accd0R49

I suppose you didn't notice this. If not, feel free to reopen this.

**Comment by jboxman-rh** [comment]
@rh-max, so this is 4.6, NOT 4.7, and 4.8+?

Thanks!

**Comment by cardil** [comment]
/hold

Until we figure out the correct approach

**Comment by rh-max** [comment]
@jboxman-rh Please hold with the merge, thanks!

**Review by cardil — APPROVED** [review]
This PR is now unrelated https://issues.redhat.com/browse/SRVOCF-478

We can proceed with merges. The https://issues.redhat.com/browse/SRVOCF-478 will be resolved in different PR.

/unhold
/lgtm
from QE

**Comment by rh-max** [comment]
@jrangelramos Per @cardil 's request, I reverted:

1. the inclusion of the S2I that you suggested in https://github.com/openshift/openshift-docs/pull/51030#pullrequestreview-1130257869 
2. the reordering of the steps
3. (the buildpacks info)

Can the two of you decide on whether the S2I step should be re-added and the step order should be updated as Jefferson originally requested? Thanks.

**Comment by jrangelramos** [comment]
yea can remove just (3.) (the buildpacks info)
and retain
1. the inclusion of the S2I that you suggested in (as first item)
2. deploy task as second item

**Comment by jrangelramos** [comment]
better now!
/lgtm

**Comment by pneedle-rh** [comment]
LGTM!

**Pull request merged**